src/main/java/com/jhuguet/sb_taskv1/app/services/impl/UserServiceImpl.java

```diff
@@ -117,40 +116,62 @@ private Tag countTagIterations(Order order) throws IdNotFound, NoTagInOrder {
 
     private Map<Integer, Integer> mapCurrentTags() {
         Map<Integer, Integer> currentTags = new HashMap<>();
-        tagRepository
-                .findAll()
-                .forEach(t -> currentTags.put(t.getId(), 0));
+        tagRepository.findAll().forEach(t -> currentTags.put(t.getId(), 0));
         return currentTags;
     }
 
     private Integer getMaxTagID(Map<Integer, Integer> tags) {
-        return tags
-                .entrySet()
-                .stream()
-                .max(Map.Entry.comparingByValue())
-                .get()
-                .getValue();
+        return tags.entrySet().stream().max(Map.Entry.comparingByValue()).get().getValue();
     }
 
     @Override
-    public User signIn(User user) throws MissingEntity, MissingRequiredFields {
+    public void signIn(User user) throws MissingEntity, MissingRequiredFields, UnqualifiedAuthority {
         if (!Objects.isNull(user)) {
             userEntityValidations(user);
-            userRepository.save(user);
+            userRepository.save(assembleUserToSave(user));
         } else {
             throw new MissingEntity();
         }
-        return null;
     }
 
-    private void userEntityValidations(User user) throws MissingRequiredFields {
-        if (user
-                .getPassword()
-                .equals("") || user
-                .getUsername()
-                .equals("")) {
+    private void userEntityValidations(User user) throws MissingRequiredFields, UnqualifiedAuthority {
+        if (user.getPassword().equals("") || user.getUsername().equals("")) {
             throw new MissingRequiredFields();
         }
+
+        if (user.getUsername().equalsIgnoreCase("administrator")) {
```

Comment:
Is The 'administrator' a hardcoded username of the admin?

Reply:
It is assumed to be so, because the admin can only be inserted from database and should be a single admin, shouldn't it?

Reply:
Yes, it should, but it's a bad practice to hardcode a username. Let's check by role, not by name "administrator".

Reply:
Okay

Reply:
But if its validation for user creation, why do we add a special case for the username "administrator"? We can check all usernames from the database and if current username already exist, throw exception.

Reply:
Ah, that's just because of how the app is built as of now, but you're correct